new branch after renaming project to Classmate

liveyourheart opened the pull request.

liveyourheart committed:
new branch after renaming project to Classmate

symbyte commented:
Small bug: if i say that im a teacher and press sign up and then press the back button, the student id will be asked for without the student button being checked. You probably want to do a check and see if the student check box is checked when the fragment loads and see if you should load the student id entry part in addition to the other checks you have.

symbyte commented:
Also, if i click on the part that says name label and id label (edit: anywhere on the screen actually. I have already pushed a fix for this to this pull request though) the app immediatly crashes

symbyte commented:
The last thing I notice is that you need to update your jdk version. The fact that your jdk is 1.7 while we are all using 1.8 is causing certain files to have to be regenerated and commited every time the build is done on a computer that isnt yours.

symbyte committed:
change gradle compiled dependencies to strict versions

 This is done to prevent the regeneration of certain project files when someone with a newer version of the apk is running a build from source. It is also best practice because it ensures that every build we do across each of our systems will have the exact same library versions.

symbyte commented:
Last think wasn't last thing sorry I lied

Please always include a description of everything that is included in your pull request when creating it. It gives you a nice space to do it and everything!

Commit:
fix crash when clicking anywhere on screen

There was, for some reason, an on click handler assigned to the FrameLayout in fragment_login. The method specified in this handler was never implemented, causing the app to crash from an exception whenever the screen (except the clickable things like edit texts and checkboxes and buttons) was clicked.

symbyte commented:
Also correct the hard coded strings that are in the login layout file. I think that covers everything. I'm going to go to bed and check to see if this is ready to merge in the morning. Thanks for getting this ready so quickly!

liveyourheart committed:
updated jdk to 1.8

liveyourheart committed:
updated jdk to 1.8

liveyourheart committed:
updated autogenerated files to fit project

liveyourheart committed:
updated autogenerated files to fit project a second time

liveyourheart committed:
Merge branch 'master' of https://github.com/colegarien/SDDTermProject into mm_classmate

liveyourheart committed:
fixed bug that showed student id when going back from teacher interface fragment

liveyourheart committed:
hardcoded strings in login layout

symbyte commented:
Why did you refert my change to the build.gradle file in commit 39a941b? That change was made for a reason, which I spelled out explicitly in the commit message. If you have a good reason for making a change like that, you have to put that reasoning in the commit message. I am pushing a commit to change it back before I merge your branch in.

Commit:
re applying strict library versions

Don't change this. Having variable versioning active on libraries can lead to unrepeatable builds.

symbyte commented:
As a closing note. Never, never, never do any kind of merge action with master. What you had the same functional result as rebasing so there isn't much harm done but merging master violates the often stated rule of "do not interact with master". It also clutters up the commit history with a bunch of unneeded commits. Whenever you need to catch your branch back up with master, always rebase. No exceptions. I will gladly help you with the process if you need it. If the rebase needs to be done on a branch that you have already submitted a pull request for (like this one) I will handle the rebase myself, because there are a few extra steps you have to do to get it to work (steps that are, frankly, pretty scary). I suspect you might have run into this if you tried to do the rebase yourself, couldn't get it to work, and then did the merge instead. If you ever find yourself feeling like you nee to megre anything ever, always run it by me or @colegarien. The odds are good that a megre will never be the right move.

symbyte commented:
The code tests well. Im going to push another commit to make your login fragment the default launched fragment and bypass the debug fragment for now.

Commit:
make login fragment launch by default

The login fragment is now our default launch fragment to give the feel of a real, cohesive application. The debug fragment is still available for anyone who feels they need it. It can be accessed but clicking the Debug item in the setting menu (the three dots at the top right corner of the application)
